Documentation/specs/runtime-configuration-file.md

```diff
@@ -116,9 +123,12 @@ The files are both JSON files stored in UTF-8 encoding. Below are sample files.
 ### `runtimeOptions` Section (`.runtimeconfig.json`)
 
 This section is copied verbatim from an identical section in the input `project.json` file (with the exception of the `target` parameter which is generated by the compilation process). The `runtimeConfig` section specifies parameters to be provided to the runtime during initialization. Known parameters include:
-
-* `gcServer` - Boolean indicating if the server GC should be used (Default: _TBD_). Note: This is designed to mirror the existing [app.config](https://msdn.microsoft.com/en-us/library/ms229357.aspx) setting)
-* `gcConcurrent` - Boolean indicating if background garbage collection should be used (Default: _TBD_). Note: This is designed to mirror the existing [app.config](https://msdn.microsoft.com/en-us/library/yhwwzef8.aspx) setting).
+* `configProperties` - Indicates configuration properties that are available during the execution of the application.
+  * Examples:
+    * Full list of [configuration properties](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md) for CoreCLR.
+    * `System.GC.Server` (old: `gcServer`) - Boolean indicating if the server GC should be used (Default: `true`).
+    * `System.GC.Concurrent` (old: `gcConcurrent`) - Boolean indicating if background garbage collection should be used (Default: `false`).
+    * `Fabrikam.App.TimeoutInSeconds` (_TBD_) - An application specified property that is available in managed code, during the execution of the application.
```

Comment:
We added AppContext.GetData to enable passing things from the host to the application like APP_CONTEXT_DEPS_FILES but I wasn't aware we are planning to expose a full configuration system through here. Correct me if I'm wrong but the full framework doesn't expose such properties through AppDomain.GetData today does it? I would be a little hesitant to just add this without a scenario because it will end up getting abused as the new application configuration model and people will start asking for features which we never really intended to support here. 

cc @AlexGhiondea